Week-1/assignment1/readme.md

```diff
@@ -0,0 +1,60 @@
+4.  Networking: for network calls such as HTTP requests, using different implementations for different platform behind a platform-independent interface.
+5.  UI backend: used for drawing basic widgets like combo boxes and windows.
+6.  JavaScript interpreter. Used to parse and execute JavaScript code.
+7.  Data storage. This is a persistence layer. The browser may need to save all sorts of data locally, such as cookies.
+
+![highlevelComponent](images/highlevelComponent.png)
+
+### C - Rendering engine and its use.
+
+Explanation C - The responsibility of the rendering engine is well... Rendering, that is display of the requested contents on the browser screen. Rendering engine use: - By default, the rendering engine can display HTML and XML documents and images. It can display other types of data via plug-ins or extension; for example, displaying PDF documents using a PDF viewer plug-in. However, in this chapter we will focus on the main use case: displaying HTML and images that are formatted using CS
+
+![renderingEngine](images/renderingEngine.png)
+
+### D - Parsers (HTML, CSS, etc)
+
+Explanation D - a parser is a program, usually part of a compiler, that receives input in the form of sequential source program instructions, interactive online commands, markup tags, or some other defined interface and breaks them up into parts that can be used by other components in programming. Parsers usually divide the work between two components: the lexer (sometimes called tokenizer) that is responsible for breaking the input into valid tokens, and the parser that is responsible for constructing the parse tree by analyzing the document structure according to the language syntax rules. The lexer knows how to strip irrelevant characters like white spaces and line breaks.
+
+![parser](images/parser.png)
+
+### E- Script Processors
+
+Explanation - The Script processor allows you to specify your own processor logic for a simple processor using JavaScript or Groovy. The script is entered as an option on the script processor. use Use the script processor to define some simple processing logic that cannot be achieved easily using any of the provided processors.
+
+### F - Tree construction
+
+Explanation – Rendering process has three tree construction first is DOM Tree then CSSOM Tree then Render Tree which consist of DOM and CSSOM Trees. DOM Tree – When the browser reads HTML code, whenever it encounters an HTML element like html, body, div etc., it creates a JavaScript object called a Node. Eventually, all HTML elements will be converted to JavaScript objects. After the browser has created Nodes from the HTML document, it has to create a tree-like structure of these node objects. Since our HTML elements in the HTML file are nested inside each other, the browser needs to replicate that but using Node objects it has previously created. This will help the browser efficiently render and manage the webpage throughout its lifecycle.
+
```

Comment:
You have not mentioned what specific part of browser constructs DOM Tree and CSS Tree.

Of course, the image answers this question. But please mention it in text also.